source/client-side-encryption/tests/README.rst

```diff
@@ -0,0 +1,220 @@
+============================
+Client Side Encryption Tests
+============================
+
+.. contents::
+
+----
+
+Introduction
+============
+
+This document describes the format of the driver spec tests included in the JSON
+and YAML files included in this directory.
+
+Additional prose tests, that are not represented in the spec tests, are described
+and MUST be implemented by all drivers.
+
+TODO
+----
+Once limitations of query parsing in mongocryptd are known, add tests to exercise those limitations.
+
+Spec Test Format
+================
+
+The spec tests are closely based on the transactions spec tests and CRUD spec tests, with the following notable differences:
+
+- An ``encryption_setup`` section, describing client side encryption setup.
+
+- The addition of ``clientSideEncryptionOptions`` on clients.
+
+- The addition of the per-test ``kmsFailpoint`` and ``mongocryptdFailpoint`` to control the mock KMS an mock mongocryptd servers.
+
+- Operations may be specified per collection through an optional "object" field. I.e. operations use the default collection specified at the root of the test:
+
+  .. code:: json
+
+     {
+         "name": "insertOne",
+         "documents": [ { "_id": 1 } ]
+     }
+
+  inserts on the default collection, but
+
+  .. code:: json
+
+      {
+          "name": "insertOne",
+          "object": { "database": "db", "collection": "other" }
+          "documents": [ { "_id": 1 }]
+      }
+
+  inserts in the ``db.other`` collection.
+
+- The addition of the ``mongocryptd_expectations``, for APM events coming from mongocryptd.
+
+
+Each YAML file has the following keys:
+
+- ``database_name`` and ``collection_name``: The default database and collection
+  to use in operations. This can be overridden by a per operation ``object``.
```

Comment:
I'd propose breaking these into separate bullets for consistency with other spec formats.